ankitkr8540/LeetBragAI
# Feature/signup

to test:
try a couple of wrong inputs to check the error messages, try correct inputs and register and check for sign in

email validation api used https://docs.mailcheck.ai/reference/email-endpoint
once sign up or login successfully, will be redirected to /username page with the corresponding leetcode username

user is not allowed to enter empty space in fields
password has to be >= 12 characters
email address has to be valid based on api response
leetcode username has to exist

username: test5
pw: 123412341234
<img width="412" alt="Screenshot 2024-06-17 at 1 32 30 AM" src="https://github.com/ankitkr8540/LeetBragAI/assets/114844285/2a969fae-1b89-4814-ab8c-25a6ceea4929">
<img width="400" alt="Screenshot 2024-06-17 at 1 33 36 AM" src="https://github.com/ankitkr8540/LeetBragAI/assets/114844285/dee779ca-2905-4966-a18a-c20bb0307413">
<img width="400" alt="Screenshot 2024-06-17 at 1 35 58 AM" src="https://github.com/ankitkr8540/LeetBragAI/assets/114844285/cad2ead7-424a-44ec-abc6-ba1975b33801">
<img width="394" alt="Screenshot 2024-06-17 at 1 32 45 AM" src="https://github.com/ankitkr8540/LeetBragAI/assets/114844285/4039bd46-13f6-46da-863c-64af70a9b799">

(ps: i created multiple test accounts to test sign up. i need help to access the db. do we have functionality to check if username already exists in the db?)
